sidecar/src/service/sidecar_server.rs

```diff
@@ -47,6 +47,18 @@ fn no_response() -> NoResponse {
     future::ready(())
 }
 
+/// The `SidecarServer` struct represents a server that handles sidecar operations.
+///
+/// It maintains a list of active sessions and a counter for each session.
+/// It also holds a reference to a `TraceFlusher` for sending trace data,
+/// and a `Mutex` guarding an optional `ManualFutureCompleter` for telemetry configuration.
+///
+/// # Fields
```

Comment:
I think it's more common (and I prefer it too) to put docs about fields inline in the struct, i.e.
```
    /// Used for sending trace data
   pub trace_flusher: Arc<TraceFlusher>
```

Reply:
No objection from me, I'll update them to be inline!